### PR TITLE
CC-2247 CC Master timeout if ZK quorum can't be established

### PR DESCRIPTION
**Pull request by ed-**
Port of #2745 to support/1.1.x

Closes CC-2247

### CODE REVIEW

**Comment by tgjonestx** [comment]
Let's hold off on a merge until Brett, William and Co decide to add this to the maintenance release.

**Comment by tgjonestx** [comment]
FYI - sent an email to the management team on 6/1 to ask them if we want to include this in a support release, and if so, which release (e.g. it's probably too late for UCSPM, 1.1.6)

**Comment by ed-** [comment]
retest this

File: isvcs/zk.go [hunk 1]
```diff
@@ -79,39 +80,73 @@ func initZK() {
...
+		select {
+		case <-halt:
+			glog.V(1).Infof("Quit healthcheck for zookeeper")
+			return nil
+		default:
+			// Try ruok.
+			ruok, err := zkFourLetterWord("127.0.0.1:2181", "ruok", time.Second*10)
+			if err != nil {
+				glog.Warningf("No response to ruok from ZooKeeper: %s", err)
+				time.Sleep(1 * time.Second)
```
Comment (achow-zenoss) [inline isvcs/zk.go]: Minor, It timed out so we may not need additional sleep?

File: isvcs/zk.go [hunk 2]
```diff
@@ -79,39 +80,73 @@ func initZK() {
...
+				glog.Warningf("No response to stat from ZooKeeper: %s", err)
+				time.Sleep(1 * time.Second)
+				continue
 			}
+
+			// If we get "This ZooKeeper instance is not currently serving requests", we know it's waiting for quorum and can at least note that in the logs.
+			glog.V(2).Infof("stat: \"%s\"", stat)
+			if string(stat) == "This ZooKeeper instance is not currently serving requests\n" {
+				glog.Warningf("ZooKeeper is running, but still establishing quorum.")
+				time.Sleep(1 * time.Second)
```
Comment (achow-zenoss) [inline isvcs/zk.go]: We have time.sleep 1 sec for all continuation, should this be moved to outer loop (e.g. first or last statement of the for loop block)?

**Pull request merged**